### PR TITLE
Gitlab Enterprise documentation

### PR DESCRIPTION
**Pull request by n2ygk**
## Fleshes out Gitlab OAuth documentation

documents the Gitlab callback URL but also the various scopes that seem appropriate.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/auth-backend/README.md [hunk 1]
```diff
@@ -52,9 +52,27 @@ export AUTH_GITHUB_ENTERPRISE_INSTANCE_URL=https://x
 
 ### Gitlab
 
+#### Creating a Gitlab Enterprise OAuth application
```
Comment (stefanalund) [inline plugins/auth-backend/README.md:55]: Nitpick: please CamelCase GitLab

Comment (n2ygk) [reply plugins/auth-backend/README.md:55]: @stefanalund The current documentation style is inconsistent w.r.t. CamelCase for GitHub too as you can see [here](https://github.com/spotify/backstage/blame/master/plugins/auth-backend/README.md#L28-L30)

```text
### Github

#### Creating a GitHub OAuth application
```

So this seems like a more global style decision for the project team to make and doesn't belong as part of
this PR which is limited only to documenting how to use GitLab for OAuth.

My updates currently follow the style that was used for GitHub. I'm happy to follow whatever style is already there!

Furthermore, I am currently unable to get `yarn run lint:docs` to run locally in order to resolve this before pushing again:
```
yarn run lint:docs
yarn run v1.22.1
$ node ./scripts/check-docs-quality
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x16eb9cd]

goroutine 19226 [running]:
github.com/errata-ai/vale/lint.(*Linter).lintFiles.func1.1(0x7ffeefbfe2e5, 0x0, 0x0, 0x0, 0x1a48200, 0xc001f30f60, 0xc00012b320, 0x1824700)
        github.com/errata-ai/vale/lint/lint.go:126 +0x4d
path/filepath.Walk(0x7ffeefbfe2e5, 0x0, 0xc000116f78, 0x208f9c0, 0x208f9c0)
        path/filepath/path.go:404 +0x6b
github.com/errata-ai/vale/lint.(*Linter).lintFiles.func1(0x7ffeefbfe2e5, 0x0, 0xc000ae6300, 0xc00012b2c0, 0xc00012a180, 0xc001c69500)
        github.com/errata-ai/vale/lint/lint.go:124 +0x114
created by github.com/errata-ai/vale/lint.(*Linter).lintFiles
        github.com/errata-ai/vale/lint/lint.go:121 +0xb6
Language linter (vale) generated errors. Please check the errors and review any markdown files that you changed.
  Possibly update .github/styles/vocab.txt to add new valid words.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
I do note that both `github` and `Github` are in .github/styles/vocab.txt but only `Gitlab` is there so I'll try adding that and pushing up.

I'll try to circle back to debug the local golang segfault unless someone has suggestions for a fix.

**Comment by n2ygk** [comment]
I would appreciate confirmation that the scopes I've documented are all required. I'm not sure if they are all necessary. I have just begun experimenting with backstage and haven't done anything with git yet.

**Review by Rugvip — APPROVED** [review]
Yep, scopes look alright, much of the ongoing work assumes that we have access to create PRs and so on. GitLab docs are a bit unclear to me, but I'm assuming that those permissions are required for that.

Let's fix the GitHub GitLab casing in a separate PR, since it's all over those docs anyway.

Thanks! :grin:

**Pull request merged**